analyzer.go

```diff
@@ -56,18 +65,24 @@ func (a *Analyzer) Analyze() (platform.AnalyzedMetadata, error) {
 		}
 	}
 
-	if a.restoresLayerMetadata() {
-		cacheMeta, err = retrieveCacheMetadata(a.Cache, a.Logger)
-		if err != nil {
-			return platform.AnalyzedMetadata{}, err
-		}
+	cacheMeta, err = retrieveCacheMetadata(a.Cache, a.Logger)
+	if err != nil {
+		return platform.AnalyzedMetadata{}, err
+	}
 
+	if a.restoresLayerMetadata() {
 		useShaFiles := true
 		if err := a.LayerMetadataRestorer.Restore(a.Buildpacks, appMeta, cacheMeta, NewLayerSHAStore(useShaFiles)); err != nil {
 			return platform.AnalyzedMetadata{}, err
 		}
 	}
 
+	if cacheMeta.BOM.SHA != "" {
```

Comment:
Should we also check the platform API here?

Reply:
Yes we should. This was an oversight by me. _Fixed._